### PR TITLE
Feature/switch

### PR DESCRIPTION
**Pull request by TahaNamdar**

### CODE REVIEW

File: src/components/switch/content/Switch.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { SwitchProps } from './switch_types';
+import { Label, ToggleButton, Slider, Wrapper } from '../styles/Switch.styles';
+
+const Switch: React.FC<SwitchProps> = ({ ...rest }: SwitchProps) => {
+  const [checked, setChecked] = React.useState<boolean>(rest.checked);
```
Comment (alirezatahani0) [inline src/components/switch/content/Switch.tsx]: mostaghim props ro nakon too state ghalate in kar

File: src/components/switch/content/switch_types.d.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+
+export type SwitchProps = {
+  children?: React.ReactNode;
+  style?: React.CSSProperties;
+  onChange?: React.ChangeEventHandler;
+  disabled?: boolean;
+  checked?: boolean;
+  size?: 'sm' | 'md';
```
Comment (alirezatahani0) [inline src/components/switch/content/switch_types.d.ts:9]: baghie size ha chi shodan pas

File: src/components/switch/content/switch_types.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  label?: string;
+  color?: 'primary' | 'secondary' | 'danger' | 'success' | 'warning';
+  value?: 'top' | 'start' | 'bottom' | 'end';
```
Comment (alirezatahani0) [inline src/components/switch/content/switch_types.d.ts]: in chie ??

File: src/components/switch/styles/Switch.styles.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import styled from 'styled-components';
+import { SwitchProps } from '../content/switch_types';
+import { SizeBtn, SizeSlider, position } from '../utils/constants';
```
Comment (alirezatahani0) [inline src/components/switch/styles/Switch.styles.ts]: need empty space after imports

File: src/components/switch/styles/Switch.styles.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+export const Wrapper = styled.div<SwitchProps>(({}) => ({
+  position: 'relative',
+}));
+
+export const Label = styled.div<SwitchProps>(({ theme, value }) => ({
+  ...theme.typography.body1,
+  ...position[value],
+  position: 'absolute',
```
Comment (alirezatahani0) [inline src/components/switch/styles/Switch.styles.ts]: chera absolute ?

File: src/components/switch/styles/Switch.styles.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+}));
+
+export const ToggleButton = styled.div<SwitchProps>(
+  ({ checked, size, color, theme }) => ({
+    ...SizeBtn[size],
```
Comment (alirezatahani0) [inline src/components/switch/styles/Switch.styles.ts:13]: in chie ?

File: src/components/switch/styles/Switch.styles.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    backgroundColor: checked ? theme.palette[color].main : 'wheat',
+    margin: '10px 0',
+    borderRadius: 30,
+    display: 'flex',
+    alignItems: 'center',
+    cursor: 'pointer',
+    position: 'relative',
```
Comment (alirezatahani0) [inline src/components/switch/styles/Switch.styles.ts:20]: chera relative?

**Review by alirezatahani0 — CHANGES_REQUESTED** [review]

File: src/components/switch/content/Switch.tsx (hunk 1 above)
Comment (TahaNamdar) [reply src/components/switch/content/Switch.tsx]: khb age kar bar checked= true bezne bayad dokmeye ma roshan she hata ba refresh chejuri handle konam?

File: src/components/switch/content/switch_types.d.ts (hunk 2 above)
Comment (TahaNamdar) [reply src/components/switch/content/switch_types.d.ts:9]: ok krdam

File: src/components/switch/styles/Switch.styles.ts (hunk 6 above)
Comment (TahaNamdar) [reply src/components/switch/styles/Switch.styles.ts:13]: ye toggle button ye dayere sefid dakheleshe in bara one

File: src/components/switch/content/Switch.tsx (hunk 1 above)
Comment (TahaNamdar) [reply src/components/switch/content/Switch.tsx]: in kari ke gofti kardam kar nakard

**Pull request closed without merging**